optimum/exporters/tasks.py

```diff
@@ -863,24 +863,39 @@ def get_model_from_task(
 
     @staticmethod
     def get_exporter_config_constructor(
-        model_type: str, exporter: str, task: str = "default", model_name: Optional[str] = None
+        model: Union["PreTrainedModel", "TFPreTrainedModel"],
+        exporter: str,
+        task: str = "default",
+        model_type: Optional[str] = None,
+        model_name: Optional[str] = None,
     ) -> ExportConfigConstructor:
         """
         Gets the `ExportConfigConstructor` for a model type and task combination.
 
         Args:
-            model_type (`str`):
-                The model type to retrieve the config for.
+            model (`Union[PreTrainedModel, TFPreTrainedModel]`):
+                The instance of the model.
             exporter (`str`):
                 The exporter to use.
             task (`str`, *optional*, defaults to `"default"`):
                 The task to retrieve the config for.
+            model_type (`str`):
+                The model type to retrieve the config for.
             model_name (`Optional[str]`, *optional*):
                 The name attribute of the model object, only used for the exception message.
 
         Returns:
             `ExportConfigConstructor`: The `ExportConfig` constructor for the requested backend.
         """
+        model_type = getattr(model.config, "model_type", model_type)
+
+        if model_type is None:
+            raise ValueError("Model type cannot be inferred. Please provide the model_type for the model!")
+
+        model_type = model_type.replace("_", "-")
+        if model_name is None:
+            model_name = getattr(model, "name", None)
```

Comment:
To follow the same "pattern" as `model_type`:
```suggestion
        model_name = getattr(model, "name", model_name)
```